### DISCUSSION TITLE
Bug in Brouter offline calculation

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#203] [Q&A] [answered]
No matter if I set the option ignore_barrier or not, I always get a 60 seconds timeout when trying to add a sp to this tour.

[60 sec timeout.zip](https://github.com/devemux86/cruiser/files/11652112/60.sec.timeout.zip)

**Comment by devemux86** [comment]
What profile and options do you use?

I imported your route and with default BRouter options (image 1), I pressed "Recalculate route" and in seconds it was recalculated:

<img src="https://github.com/devemux86/cruiser/assets/3484020/cdec00ab-2611-41ae-9083-212fe874f044" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/586db042-59af-4246-b236-ad16a26e42e8" width="50%"/>

**Reply by DolomitesTom** [reply]
I see what the problem is: it is avoid toll roads. 
Avoid is normally not forbit. Avoid normally means don't use **if possible**. If you have no other chance, then use it.
But Brouter calculation seems to take this as a strict prohibition and instead of coming up with a warning message it runs in a timeout.

**Comment by devemux86** [comment]
> Brouter calculation seems to take this as a strict prohibition

Strict avoidances in BRouter have been removed in latest Cruiser, see https://github.com/devemux86/cruiser/discussions/188#discussioncomment-6058359
You use Cruiser's internal profiles, right?

In your example there is a border crossing with tolls.
If we enable avoid tolls, even in Kurviger we get this strange result: https://kurv.gr/2dukK

We can make the avoidance in BRouter less strict, e.g. change "1" to "100" in line 190:
```
  min if is_avoided_toll_road then 100 else 999
```
Then the BRouter routing works with a similar result.

I already plan to review the BRouter options: add more avoid options and strength of avoidances, make them less strict, etc.

**Reply by DolomitesTom** [reply]
Sounds like a good idea. Then you get at least a calculated route and not a timeout.

**Comment by devemux86** [comment, marked as answer]
It should work properly with the new Cruiser modified driving BRouter profile available in [Cruiser 3.0.23](https://play.google.com/store/apps/details?id=gr.talent.cruiser):

- https://github.com/devemux86/cruiser/discussions/175#discussioncomment-6161554
